servers/rendering/renderer_rd/renderer_scene_render_rd.h

```diff
@@ -94,7 +94,7 @@ class RendererSceneRenderRD : public RendererSceneRender {
 	double time_step = 0;
 
 	struct RenderBufferData {
-		virtual void configure(RID p_color_buffer, RID p_depth_buffer, int p_width, int p_height, RS::ViewportMSAA p_msaa, uint32_t p_view_count) = 0;
+		virtual void configure(RID p_color_buffer, RID p_depth_buffer, RID p_target_buffer, int p_width, int p_height, RS::ViewportMSAA p_msaa, uint32_t p_view_count) = 0;
```

Comment:
Keep in mind you cant always assume a target buffer for subpasses, because it is perfectly possible to render 3D at say half resolution for performance, so assume this will not always be the case.

Reply:
Indeed, that is something we need to add. I'm passing it here so we can use it in the situation where we need it.

This is something I don't really like about the current setup, things are weirdly split between RendererSceneRenderRD and its implementations. There is too much assumed that it will only work one way, when we introduce different methods it gets really messy passing things around we only need "sometimes".

IMHO it should be the subclass that creates the images and buffers it needs, not the super class. That means a little bit of duplication of code between the two implementations but we're already talking about only a few lines of code that is starting to get more and more different and we're now jumping through hoops to implement the differences.

Reply:
I guess that sounds ok, feel free to change this if you want then.

Reply:
> I guess that sounds ok, feel free to change this if you want then.

@reduz  for now leaving it as is, it works fine right now and moving the code is just a pretty up thing. I think its best if you play around with this after its merged and see where we go from there.